lib/data/entity.js

```diff
@@ -88,6 +88,35 @@ const parseSubmissionXml = (entityFields, xml) => new Promise((resolve, reject)
   });
 });
 
+const parseCreateJson = (body, properties, datasetName) => new Promise((resolve, reject) => {
+  const entity = { system: Object.create(null), data: Object.create(null) };
+
+  try {
+    const { uuid, create, data } = body;
+
+    entity.system.dataset = datasetName;
+    entity.system.id = uuid;
+    entity.system.create = create;
+
+    const propSet = new Set(properties.map(p => p.name));
+    for (const [k, v] of Object.entries(data)) {
+      if (k === 'label') entity.system.label = v;
+      else if (propSet.has(k))
+        entity.data[k] = isBlank(v) ? '' : v;
+      else
+        throw Problem.user.invalidEntity({ reason: `You specified the dataset property [${k}] which does not exist.` });
+    }
```

Comment:
these lines are duplicated with `parseUpdateJson`, can be extracted to a new function